### PR TITLE
Add compatibility with imagine-svg 1.0

### PR DESCRIPTION
**Pull request by ausi**
~~Depends on the release of `contao/imagine-svg 1.0.0`~~  
Released: https://github.com/contao/imagine-svg/releases/tag/1.0.0

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
         "contao-components/tablesorter": "^2.1",
         "contao-components/tinymce4": "^4.7",
         "contao/image": "^1.0",
-        "contao/imagine-svg": "^0.2.3",
+        "contao/imagine-svg": "^0.2.3 || ^1.0@RC",
```
Comment (ausi) [inline composer.json]:
```suggestion
        "contao/imagine-svg": "^0.2.3 || ^1.0",
```

File: core-bundle/composer.json [hunk 2]
```diff
@@ -45,7 +45,7 @@
         "contao-components/tablesorter": "^2.1",
         "contao-components/tinymce4": "^4.7",
         "contao/image": "^1.0",
-        "contao/imagine-svg": "^0.2.3",
+        "contao/imagine-svg": "^0.2.3 || ^1.0@RC",
```
Comment (ausi) [inline core-bundle/composer.json]:
```suggestion
        "contao/imagine-svg": "^0.2.3 || ^1.0",
```

**Review by aschempp — DISMISSED** [review]

File: core-bundle/src/Resources/contao/classes/DataContainer.php [hunk 3]
```diff
@@ -622,7 +622,7 @@ protected function row($strPalette=null)
 
 				$image = Image::getPath('placeholder.svg');
 
-				if ($blnCanResize)
+				if ($blnCanResize && $objFile->viewWidth && $objFile->viewHeight)
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/classes/DataContainer.php]: Why did you add this check?

Comment (ausi) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: SVG images with undefined sizes cannot be resized. I tested different SVG images in the Contao backend and found this problematic use case.

Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: Ok, then the following would be more clear IMHO:

```php
$blnCanResize = true;

if ($objFile->isSvgImage)
{
    // SVG images with undefined sizes cannot be resized
    if (!$objFile->viewWidth || !$objFile->viewHeight)
    {
        $blnCanResize= false;
    }
}
elseif (System::getContainer()->get('contao.image.imagine') instanceof Imagine)
{
    // Check the maximum width and height if the GDlib is used to resize images
    if ($objFile->height > Config::get('gdMaxImgHeight') || $objFile->width > Config::get('gdMaxImgWidth'))
    {
        $blnCanResize = false;
    }
}
```

WDYT?

Comment (ausi) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: Yes, your version is way more clear! Can you commit directly to my PR?

Comment (leofeyer) [reply core-bundle/src/Resources/contao/classes/DataContainer.php]: Done in ef19111f6ac915aff02ea640176679a94c042892.

**Review by leofeyer — APPROVED** [review]

**Review by ausi — COMMENTED** [review]
✅ Approved

**Comment by leofeyer** [comment]
Thank you @ausi.
